software/contrib/preset_manager.py

```diff
@@ -0,0 +1,191 @@
+        
+        oled.text(name, int((OLED_WIDTH - (len(name) * CHAR_WIDTH)) / 2), 0, 1)
+        
+        self.update_current_value(values)
+        
+        for index, value in enumerate(values):
+            x = int((index % 3)*self.box_width)
+            y = int((index // 3)*self.box_height) + self.box_height - 1
+            oled.rect(x, y, self.box_width, self.box_height, 1) if index != self.selected_preset else oled.fill_rect(x, y, self.box_width, self.box_height, 1)
+            oled.text(str(value), (x + 2), (y + 2), 1 if index != self.selected_preset else 0)
+        oled.show()    
+        
+
+    def update_outputs(self):
+        values = self.presets[self.loaded_preset][1]
+        for cv_value in zip(cvs, values):
+            cv_value[0].voltage((cv_value[1] * self.output_multiplier))
+
+
+    def draw_preset_list(self):
+        oled.fill(0)
+        
+        positions = len(self.presets) + 1
+        self.hovered_preset = k1.read_position(positions)
+        
+        vertical_offset = -(self.hovered_preset * self.list_line_height) + 1
+        oled.fill_rect(0, 0, OLED_WIDTH, self.list_line_height, 1)
+        for index, preset in enumerate(self.presets):
+            name = preset[0]
+            oled.text(name, 0, vertical_offset, 1 if self.hovered_preset != index else 0)
+            
+            vertical_offset += self.list_line_height
+        if self.hovered_preset == (positions - 1):
+            colour = 0
+            self.create_new_preset = True
+        else:
+            colour = 1
+            self.create_new_preset = False
+        oled.text('NEW PRESET', 0, vertical_offset, colour)
+        oled.show()
+    
+
+    def load_new_preset(self):
+        if self.create_new_preset:
+            self.presets.append(self.blank_preset)
+            self.loaded_preset = -1
+        else:
+            self.loaded_preset = self.hovered_preset
+
+
+    def draw_rename_screen(self):
+        oled.fill(0)
+        
+        name = self.presets[self.hovered_preset][0]
+        name_list = list(name)
+        while len(name_list) < self.max_characters:
+            name_list.append(' ')
+        
+        old_selected_letter = self.selected_letter
+        self.selected_letter = k1.read_position(self.max_characters)
+        if old_selected_letter != self.selected_letter:
+            self.old_k2 = k2.read_position(len(self.characters) + 1)
+        else:
+            if k2.read_position(len(self.characters) + 1) != self.old_k2:
+                name_list[self.selected_letter] = k2.choice(self.characters)
+                self.presets[self.hovered_preset][0] = ''.join(name_list)
+        
+        oled.fill_rect((self.selected_letter * CHAR_WIDTH), 0, CHAR_WIDTH, (CHAR_HEIGHT + 2), 1)
+        for index, letter in enumerate(name_list):
+            oled.text(letter, (index * CHAR_WIDTH), 1, 0 if self.selected_letter == index else 1)
+        
+        oled.text('< DELETE', 0, 24, 1)
+
+
+    def hover_delete(self):
```

Comment:
Is this something enough other scripts could use that it's worth refactoring up to the parent?

Reply:
The delete text you mean, or the whole delete process?

Reply:
The whole delete dialog. It looks like it's laid out to fit the buttons. If there's any other script that allows you to delete something, it could be handy to have a dialog to reuse.